technico-v16/app/components/modal.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import Link from 'next/link';
+import { URLS } from '../lib/constants';
+
+type UserResponse = {
+  id: string;
+  name?: string;
+  surname?: string;
+};
+
+const ModalComponent: React.FC = () => {
+  const [users, setUsers] = useState<UserResponse[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<string | null>(null);
+
+  const openModal = async () => {
+    const modal = document.getElementById('my_modal_1') as HTMLDialogElement;
+    if (modal) {
+      modal.showModal();
+      await fetchUsers();
+    }
+  };
+
+  const fetchUsers = async () => {
+    try {
+      const response = await fetch('https://localhost:7118/api/Users');
+      if (!response.ok) {
+        throw new Error(`Error ${response.status}: Failed to fetch users`);
+      }
+      const data: UserResponse[] = await response.json();
+      setUsers(data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(event.target.value);
+  };
+
```

Comment:
-  Good use of loading state display
-  Good use of disabled state on select's default option